# CNV estimation from DNA methylation sequencing data (RRBS)

This tutorial is based on [HMMcopy R package vignette](https://www.bioconductor.org/packages/release/bioc/html/HMMcopy.html)

**Used tools:**
> HMMcopy (R package)

**Used files:** 
> Aligned reads (BAM file)

> Reference genome (hg 19)
 
> wgEncodeCrgMapabilityAlign100mer.bigWig downloaded from [UCSC Golden path](http://hgdownload.cse.ucsc.edu/goldenpath/hg19/encodeDCC/wgEncodeMapability/)
           
           
### HMMCopy R package is designed to estimate copy number variation (CNV) from whole genome sequencing data (WGS). But, here I will use it to estimate CNV from Reduced Representative Bisulfite seqencing data -DNA Methylation-.
****************

HMMcopy consists of 2 parts:
1. Terminal based tools
2. R package
**************

## First, 3 wig files should be generated from "terminal based" tools of hmmcopy_util:
*follow the link to install the tools : [hmmcopy_utils](https://github.com/shahcompbio/hmmcopy_utils)*

1. Wig contains the read counts from the BAM file by using readCounter tool (-w determines the size of the window/bin, here I used 10kb) :
```
./readCounter -w 10000 sample.sorted.bam > sample.wig
```
2. Wig contains the GC content from reference geneome (fasta)
```
./gcCounter -w 10000 sample.sorted.bam > gc.wig
```

3. Wig contains the mappability counts from wgEncodeCrgMapabilityAlign100mer.bigWig file 
** I this case 100mer will be used because the length of my sequencing reads is 150bp (for compatibility).
```
./mapCounter -w 10000 wgEncodeCrgMapabilityAlign100mer.bigWig > map.wig
```
********
## Now R time:
*Delete chrM from the 3 wig files (for example: manually from text editor)*
*HMMcopy is designed to plot one chr only each time (plotSegments), but the data we have is whole genome data. So, we should subset the data*
```
setwd("~/myfiles")
library(HMMcopy)
raw_data <- wigsToRangedData(readfile = "sample.wig" ,gcfile = "gc.wig" ,mapfile = "map.wig")
# subset the data, chr7 for example
raw_data <- raw_data[raw_data$chr=="chr7",]
# correct the data (it will eliminate the outlier bins)
corrected <- correctReadcount(raw_data) 
# adjust the parameters generated by hmm algorithm
parameters <- HMMsegment(corrected, getparam = TRUE)  
parameters$mu <- log(c(1, 1.4, 2, 2.7, 3, 4.5) / 2, 2) 
parameters$m <- parameters$mu
segmented_copy <- HMMsegment(corrected, parameters)
plotSegments(corrected, segmented_copy) #plot CNV graph
```




